src/map.cpp

```diff
@@ -157,11 +156,11 @@ void Map::removeTile(uint16_t x, uint16_t y, uint8_t z)
 		return;
 	}
 
-	Tile* tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
+	std::shared_ptr<Tile> tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
```

Comment:
Could this be `const auto`
```suggestion
	const auto& tile = floor->tiles[x & FLOOR_MASK][y & FLOOR_MASK];
```